cloudinit/sources/tests/test_oracle.py

```diff
@@ -418,36 +404,76 @@ class TestReadOpcMetadata:
     does_not_raise = ExitStack
 
     @pytest.fixture(autouse=True)
-    def configure_opc_metadata_in_httpretty(self, httpretty):
+    def configure_v1_metadata_in_httpretty(self, httpretty):
         """Configure HTTPretty with the various OPC metadata endpoints."""
         httpretty.register_uri(
             httpretty.GET,
             "http://169.254.169.254/opc/v1/instance/",
             OPC_V1_METADATA,
         )
 
-    def test_json_decoded_value_returned(self):
+    @pytest.fixture()
+    def with_v2(self, httpretty):
+        httpretty.register_uri(
+            httpretty.GET,
+            "http://169.254.169.254/opc/v2/instance/",
+            OPC_V2_METADATA,
+        )
+
+    @pytest.fixture()
+    def no_v2(self, httpretty):
+        httpretty.register_uri(
+            httpretty.GET,
+            "http://169.254.169.254/opc/v2/instance/",
+            status=404,
+        )
+
+    def test_json_decoded_value_returned(self, with_v2):
         # read_opc_metadata should JSON decode the response and return it
+        expected = json.loads(OPC_V2_METADATA)
+        assert expected == oracle.read_opc_metadata().instance_data
```

Comment:
Should we also be testing the `version` attribute of `read_opc_metadata`'s return here (and in the test below)?